### ISSUE TITLE
Compose Up command escaping issue on folder with spaces

### ISSUE THREAD

**Issue by junieresmat** [microsoft/vscode-docker#4291]
When using the **Compose Up** context menu option on a folder that has spaces in the name
![image](https://github.com/microsoft/vscode-docker/assets/13772585/eecc0b37-1a35-4608-93ab-853d7eb67ec3)

I get the following on the terminal windows

```
 Executing task: docker compose -f "My Folder\test-docker-compose.yaml" up -d --build 


Usage:  docker compose [OPTIONS] COMMAND

Define and run multi-container applications with Docker

Options:
      --ansi string                Control when to print ANSI control
                                   characters ("never"|"always"|"auto")
                                   (default "auto")
      --compatibility              Run compose in backward compatibility mode
      --dry-run                    Execute command in dry run mode
      --env-file stringArray       Specify an alternate environment file
  -f, --file stringArray           Compose configuration files
      --parallel int               Control max parallelism, -1 for
                                   unlimited (default -1)
      --profile stringArray        Specify a profile to enable
      --progress string            Set type of progress output (auto,
                                   tty, plain, quiet) (default "auto")
      --project-directory string   Specify an alternate working directory
                                   (default: the path of the, first
                                   specified, Compose file)
  -p, --project-name string        Project name

Commands:
  attach      Attach local standard input, output, and error streams to a service's running container
  build       Build or rebuild services
  config      Parse, resolve and render compose file in canonical format
  cp          Copy files/folders between a service container and the local filesystem
  create      Creates containers for a service
  down        Stop and remove containers, networks
  events      Receive real time events from containers
  exec        Execute a command in a running container
  images      List images used by the created containers
  kill        Force stop service containers
  logs        View output from containers
  ls          List running compose projects
  pause       Pause services
  port        Print the public port for a port binding
  ps          List containers
  pull        Pull service images
  push        Push service images
  restart     Restart service containers
  rm          Removes stopped service containers
  run         Run a one-off command on a service
  scale       Scale services
  start       Start services
  stats       Display a live stream of container(s) resource usage statistics
  stop        Stop services
  top         Display the running processes
  unpause     Unpause services
  up          Create and start containers
  version     Show the Docker Compose version information
  wait        Block until the first service container stops
  watch       Watch build context for service and rebuild/refresh containers when files are updated

Run 'docker compose COMMAND --help' for more information on a command.
unknown docker command: "compose Folder\\test-docker-compose.yaml"

 *  The terminal process "C:\Users\{MyUser}\AppData\Local\Microsoft\WindowsApps\Microsoft.PowerShell_8wekyb3d8bbwe\pwsh.exe -Command docker compose -f "My Folder\test-docker-compose.yaml" up -d --build" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.
```
It seems like the escaping is the command is causing this. 
![image](https://github.com/microsoft/vscode-docker/assets/13772585/b164e4e6-3844-4225-9b99-18f9a20d1b00)

After I rename the folder to remove the space "My Folder" => "My_Folder" the command works fine.
![image](https://github.com/microsoft/vscode-docker/assets/13772585/c7e1d654-16b7-40b0-a6e3-03db1dba59df)

**Comment by bwateratmsft** [comment]
I am able to reproduce the issue on Windows. Still need to check if this affects Mac and Linux.

**Comment by bwateratmsft** [comment]
I think we need different quoting for CMD vs. everything else.

**Comment by bwateratmsft** [comment]
So there's a couple of workarounds you can do to get around the issue.

1. Change your default terminal profile to Command Prompt (instead of pwsh). This is the easiest but it's also heavy-handed. You can do this with the command "Terminal: Select Default Profile". You can also put this setting into `.vscode/settings.json` so it applies only to that particular workspace instead of all projects.
2. You can use [command customization](https://code.visualstudio.com/docs/containers/reference#_command-customization) to write up a custom compose up / down command for use in this workspace, in `.vscode/settings.json`. Command customization is highly flexible and would be able to solve your problem.

Because we don't know what shell the user has present--VSCode doesn't give us an API to tell--we don't know what quoting we _should_ do. And because command customization is designed to take string command lines, instead of an array of arguments, we can't take advantage of VSCode's intelligence here.

This might be a won't-fix based on the workarounds and the complexity of solving it, but I'll wait until @fiveisprime can weigh in.

**Comment by bwateratmsft** [comment]
This is now released in Docker extension version 1.29.4.
